packages/dynamodb/src/index.ts

```diff
@@ -1,447 +1,436 @@
+        },
+        ExpressionAttributeValues: {
+          ":gsi1pk": `SESSION#${sessionToken}`,
+          ":gsi1sk": `SESSION#${sessionToken}`,
+        },
+      })
+      if (!data.Items || data?.Items?.length <= 0) return null
 
-            return null
-          }
+      const infoToDelete = data.Items[0]
 
-          return data.Item || null
-        },
-        async deleteVerificationRequest(identifier, token) {
-          // eslint-disable-next-line @typescript-eslint/return-await
-          return await client
-            .delete({
-              TableName,
-              Key: {
-                pk: `VR#${identifier}`,
-                sk: `VR#${hashToken(token)}`,
-              },
-            })
-            .promise()
+      const res = await client.delete({
+        TableName,
+        Key: {
+          pk: infoToDelete.pk,
+          sk: infoToDelete.sk,
         },
+        ReturnValues: "ALL_OLD",
+      })
+      const session = res.Attributes as AdapterSessionDynamo
+      return session ? unMarshallAdapterSession(session) : null
+    },
+    async createVerificationToken(verificationToken) {
+      const { identifier, expires, token } = verificationToken
+      const item: VerificationTokenDynamo = {
+        pk: `VR#${identifier}`,
+        sk: `VR#${token}`,
+        token,
+        identifier,
+        type: "VR",
+        expires: expires.getTime() / 1000,
       }
+      await client.put({ TableName, Item: item })
+      return unMarshallVerificationToken(item)
     },
+    async useVerificationToken({ identifier, token }) {
+      const data = await client.delete({
+        TableName,
+        Key: {
+          pk: `VR#${identifier}`,
+          sk: `VR#${token}`,
+        },
+        ReturnValues: "ALL_OLD",
+      })
+      const attributes = data.Attributes as VerificationTokenDynamo
+      return attributes ? unMarshallVerificationToken(attributes) : null
+    },
+  }
+}
+// these functions are used to remove dynamodb specific properties
+// and to transform back date strings into Date javascript objects
+// the unused vars are only there so that the ...rest object does not contain those dynamodb internal properties
+const unMarshallAdapterSession = (
+  session: AdapterSessionDynamo
+): AdapterSession => {
+  const { pk, sk, GSI1PK, GSI1SK, type, ...rest } = session
+  return {
+    ...rest,
+    expires: new Date(session.expires * 1000),
+  }
+}
+
+const unMarshallVerificationToken = (
+  verificationToken: VerificationTokenDynamo
+): VerificationToken => {
+  const { pk, sk, type, ...rest } = verificationToken
+  return {
+    ...rest,
+    expires: new Date(verificationToken.expires * 1000),
+  }
+}
+
+const unMarshallAdapterUserDynamo = (
+  adapterUser: AdapterUserDynamo
+): AdapterUser => {
+  const { pk, sk, GSI1PK, GSI1SK, type, ...rest } = adapterUser
+  return {
+    ...rest,
+    emailVerified: adapterUser.emailVerified
+      ? new Date(adapterUser.emailVerified)
+      : null,
   }
 }
```

Comment:
These could be simplified to a much simpler interface, see examples here:

https://github.com/nextauthjs/adapters/blob/e6915fa1dcba9b12f26bed55cddcb3b3bcd7b07c/packages/fauna/src/index.ts#L53-L80

https://github.com/nextauthjs/adapters/blob/e6915fa1dcba9b12f26bed55cddcb3b3bcd7b07c/packages/neo4j/src/utils.ts#L1-L41

Very important that you don't use the field names whenever possible, to make sure that the user can add any fields if they want to.